### DISCUSSION TITLE
Is it possible to use theads code?

### DISCUSSION THREAD

**Discussion by mrceperka** [phper-framework/phper#25] [Q&A] [answered]
Hello,

I was wondering if threads or even `async` could be used in rust code. I guess that for async, runtime is needed so -> no, but threads -> maybe? :D

I don't know rust much, sorry.

Thanks for this repository btw :pray:

**Comment by jmjoy** [comment, marked as answer]
You can spawn threads or async coroutine, spawn thread: <https://doc.rust-lang.org/std/thread/fn.spawn.html>, spawn coroutine using tokio:  <https://docs.rs/tokio/1.9.0/tokio/fn.spawn.html>.

If you are using NTS mode, spawn threads or coroutine in module init function will be persistent, spawn threads or coroutine in request lifetime will be drop after request finished.

You can reference to http server example: <https://github.com/jmjoy/phper/tree/master/examples/http-server>.

**Reply by mrceperka** [reply]
Thanks for the answer, that was fast :) I've missed that example, sorry.
